contracts/RFQ.sol

```diff
@@ -194,10 +204,17 @@ contract RFQ is IRFQ, ReentrancyGuard, SignatureValidator, BaseLibEIP712 {
 
         // Transfer token/Eth to receiver
         if (_order.makerAssetAddr == address(weth)) {
+            // Transfer from maker
+            spender.spendFromUser(_order.makerAddr, _order.makerAssetAddr, settleAmount);
             weth.withdraw(settleAmount);
             payable(_order.receiverAddr).transfer(settleAmount);
         } else {
-            IERC20(_order.makerAssetAddr).safeTransfer(_order.receiverAddr, settleAmount);
+            spender.spendFromUserTo(_order.makerAddr, _order.makerAssetAddr, _order.receiverAddr, settleAmount);
+        }
+        // Collect fee
+        // feeAmount = _order.makerAssetAmount - settleAmount;
+        if (_order.makerAssetAmount - settleAmount > 0) {
```

Comment:
Would it be more efficient to check `_order.feeFactor > 0`?

Reply:
Perhaps having `uint256 fee = _order.makerAssetAmount.mul(_order.feeFactor).div(BPS_MAX)` and set `settleAmount = _order.makerAssetAmount.sub(fee)`?

Reply:
Fixed in f884993